index.js

```diff
@@ -57,7 +58,61 @@ module.exports = {
     return new MergeTrees(addonTrees);
   },
 
-  _moduleUnificationTrees() {
+  // Trigger exception if the result of `addon.resolverConfig` method has an unexpected format
+  // Show a warning if there are collisions between addon types or addon collections
+  validateAddonsConfig: function(addonsConfig) {
+
+    let types = {};
+    let collections = {};
+
+    Object.keys(addonsConfig).forEach(addonName => {
+
+      let addonConfig = addonsConfig[addonName];
+      if (addonConfig) {
+        if (typeof(addonConfig) !== 'object') {
+          throw new Error(`"addon.resolverConfig" returns an unexpected value. Addon: ${addonName}.`);
+        }
+
+        let addonTypes = addonConfig.types || {};
+        if (typeof(addonTypes) !== 'object') {
+          throw new Error(`"addon.resolverConfig" returns an unexpected "types" value. Addon: ${addonName}.`);
+        }
+
+        let addonCollections = addonConfig.collections || {};
+        if (typeof(addonCollections) !== 'object') {
+          throw new Error(`"addon.resolverConfig" returns an unexpected "collections" value. Addon: ${addonName}.`);
+        }
+
+        Object.keys(addonTypes).forEach(key => {
+          if (!types.hasOwnProperty(key)) {
+            types[key] = key;
+          } else {
+            this.ui.writeLine(`Addon '${types[key]}' configured the type '${key}' on the resolver, but addon '${addonName}' has overwritten the type '${key}'.`);
```

Comment:
does the current resolver config pass these checks?

Reply:
This validation warns the user on the console that 2 addons are resolving the same type or collection. The `mergeAddonsConfig` function triggers an error if the `addonsConfig` is not modified. 

This behavior is still pending of confirmation.